src/components/MoneyRequestConfirmationList.js

```diff
@@ -125,18 +151,41 @@ const defaultProps = {
     receiptSource: '',
     listStyles: [],
     policyCategories: {},
+    transactionID: '',
+    transaction: {},
+    mileageRate: {unit: CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES, rate: 0, currency: 'USD'},
+    isDistanceRequest: false,
 };
 
 function MoneyRequestConfirmationList(props) {
     // Destructure functions from props to pass it as a dependecy to useCallback/useMemo hooks.
     // Prop functions pass props itself as a "this" value to the function which means they change every time props change.
-    const {onSendMoney, onConfirm, onSelectParticipant} = props;
+    const {onSendMoney, onConfirm, onSelectParticipant, transaction} = props;
     const {translate} = useLocalize();
 
     // A flag and a toggler for showing the rest of the form fields
     const [showAllFields, toggleShowAllFields] = useReducer((state) => !state, false);
     const isTypeRequest = props.iouType === CONST.IOU.MONEY_REQUEST_TYPE.REQUEST;
 
+    const {unit, rate, currency} = props.mileageRate;
+    const distance = lodashGet(transaction, 'routes.route0.distance', 0);
+    const shouldCalculateDistanceAmount = props.isDistanceRequest && props.iouAmount === 0;
+    const shouldCategoryEditable = !_.isEmpty(props.policyCategories) && !props.isDistanceRequest;
+
+    const formattedAmount = CurrencyUtils.convertToDisplayString(
+        shouldCalculateDistanceAmount ? DistanceRequestUtils.getDistanceRequestAmount(distance, unit, rate) : props.iouAmount,
+        props.isDistanceRequest ? currency : props.iouCurrencyCode,
+    );
+
+    useEffect(() => {
+        if (!shouldCalculateDistanceAmount) {
+            return;
+        }
+
+        const amount = DistanceRequestUtils.getDistanceRequestAmount(distance, unit, rate);
+        IOU.setMoneyRequestAmount(amount);
+    }, [shouldCalculateDistanceAmount, distance, rate, unit]);
+
```

Comment:
@hayata-suenaga Well, I was speaking in the context of the mentioned lines and its direct sourroudnigns, but specifically, I'm talking about this range of lines, focusing on checking if `props.iouAmount` equals `0` and the dependencies of the `useEffect` hook.

This code cause [the bug](https://github.com/Expensify/App/issues/26946) I mentioned earlier and we're trying to figure out whether we can just drop the `props.iouAmount === 0` check without causing a regression.

One option how this area could look like after the change was included in [my previous message](https://github.com/Expensify/App/pull/25707/files#r1321289175). Let's continue the discussion in this thread.

Reply:
ahh now I see what's you're talking about

we actually need that conditional statement

when the user manually edit the distance amount, we want to keep that amount instead of overwriting it with the calculated amount

but I know see why this is causing the issue

can we just clear out the amount from Transaction object when navigating back to the participants page?

Reply:
> can we just clear out the amount from Transaction object when navigating back to the participants page?
I'm not completely sure, I'll come back to you

Reply:
@hayata-suenaga 
> when the user manually edit the distance amount, we want to keep that amount instead of overwriting it with the calculated amount

Users can't edit the distance amount manually (we prevented user clicking on the amount field), they only can edit the waypoint

Reply:
@DylanDylann Is it a temporary measure, or can we (at least for now) take this for granted?

Reply:
I am not sure, @hayata-suenaga Could you help to confirm that?

Reply:
yep that is a temporary major so we still have to account for that

Reply:
@hayata-suenaga

Do we have a GH issue to re-add the ability to edit the amount manually?

We're in a tricky situation here. Currently, the simplest way to fix https://github.com/Expensify/App/issues/26946 is just dropping the `=== 0` check, taking advantage from the fact that the amount cannot be (currently) edited by the user.

If we assume that that amount can be edited (because the feature might be re-added), we'd probably have to add some new persistent state to distinguish manually edited value from the auto-calculated one, but wouldn't be able to test it properly.

There are multiple solutions to this problem, but I would prefer to do it this way: for now, let's remove the `=== 0` condition, fixing https://github.com/Expensify/App/issues/26946. Then, comment on the GH issue to re-add the manual edition feature to take care not to introduce regression for https://github.com/Expensify/App/issues/26946 (creating new persistent state, if needed).

If there's no issue for re-adding manual distance IOU amount edition, we can create it.

This way, we're be focusing on problems we have right now, but at the same time do our best not to introduce regressions. This will also do fine in a hypothetical case we actually never re-allow manual edition (but such case wouldn't surprise me).

Reply:
I rather prefer if you could handle it in your issue @cubuspl42 🙇

Reply:
@hayata-suenaga But how would you suggest to test that it achieves the goal, if it's currently not possible to manually edit the amount?

Maybe we could introduce a beta for editing the distance request amount, so this can be tested locally via manual one-line edition in `Permissions.js`, while the QA could be opted in to that beta on the backend?

Reply:
I'm not totally following everything discussed here, but I am chiming in to say that I am working on a PR https://github.com/Expensify/App/pull/26141 to let people edit the distance request, and it will allow users to edit the amount as well.

Reply:
@cubuspl42 you can hold your issue on the above PR if necessary